helpers/parse_helpers.go

```diff
@@ -139,38 +141,42 @@ func processPwdString(pwdString string) ([]byte, error) {
 // - <filename>:pass=<password>
 // - <filename>:fd=<filedescriptor>
 // - <filename>:<password>
-func processPrivateKeyFiles(keyFilesAndPwds []string) ([][]byte, [][]byte, [][]byte, [][]byte, [][]byte, error) {
+// - keyprovider:<...>
+func processPrivateKeyFiles(keyFilesAndPwds []string) ([][]byte, [][]byte, [][]byte, [][]byte, [][]byte, [][]byte, error) {
 	var (
 		gpgSecretKeyRingFiles [][]byte
 		gpgSecretKeyPasswords [][]byte
 		privkeys              [][]byte
 		privkeysPasswords     [][]byte
 		pkcs11Yamls           [][]byte
+		keyProviders          [][]byte
 		err                   error
 	)
 	// keys needed for decryption in case of adding a recipient
 	for _, keyfileAndPwd := range keyFilesAndPwds {
 		var password []byte
-		// skip for "provider" protocol
+
+		// treat "provider" protocol separately
 		if strings.HasPrefix(keyfileAndPwd, "provider:"){
+			keyProviders = append(keyProviders, []byte(keyfileAndPwd[9:]))
```

Comment:
I am not sure why we have to parse private key file, there wont be any private key that will be provided during decryption process by skopeo. All the required keys will be provided by the executable command/ grpc method, that is taken care by key provider implementation by parsing the config file.

Reply:
Because this is how all other keys are also parsing their private key files. This is where they are getting them from.

Reply:
processRecipientKeys is not used for the purpose of decryption than by pkcs7 to get x509 certs. See change below.

Reply:
Even in case of key provider implementation, we wont be passing private keys in command as per proposal https://github.com/containers/ocicrypt/issues/35 
OCICRPYT_CUSTOM_CONFIG=/etc/ocicrypt_custom.json skopeo copy --decryption-key custom:some-params
So here the executable command mentioned in /etc/ocicrypt_custom.json will be providing keys based on the protocol present in manifest. @lumjjb  correct me if I am wrong.

Reply:
I am trying to keep imgcrypt's parse_helper.go and this parse_helper.go in sync. In imgcrypt we pass all private keys with the `--key` option on the command line and have so far parsed all private keys with this function: https://github.com/stefanberger/imgcrypt/commit/5dfd1d9bfa62d3d516b92224728fb2add99e482c#diff-3ba41d08a6db765fcc9344a7ded69a7bacf55e9847d68169a2f9bb46882448ccR231

So it may not be correct that skopeo would not pick up the private key for keyprovider but in imgcrypt  we need to parse the private key, even if it's just some sort of a URI or plain string, form this function since this one is passed whatever the user provided with the `--key` option on the command line.

Reply:
Modified the title and commit text with better explanation for this change.

Reply:
ok, I was not aware of this.

Reply:
```suggestion
			keyProviders = append(keyProviders, []byte(keyfileAndPwd[len("provider:"]))
```
nit